tests/hipscat_import/cross_match/test_macauff_arguments.py

```diff
@@ -0,0 +1,354 @@
+            left_ra_column="ra",
+            left_dec_column="dec",
+            left_id_column="id",
+            right_catalog_dir=small_sky_source_catalog,
+            right_ra_column="source_ra",
+            right_dec_column="source_dec",
+            right_id_column="source_id",
+            input_path=small_sky_dir,
+            input_format="csv",
+            metadata_file_path=formats_yaml,
+        )
+
+
+def test_macauff_args_no_right_catalog(
+        small_sky_object_catalog,
+        small_sky_dir,
+        formats_yaml,
+        tmp_path
+):
+    with pytest.raises(ValueError, match="right_catalog_dir"):
+        args = MacauffArguments(
+            output_path=tmp_path,
+            output_catalog_name="object_to_source",
+            tmp_dir=tmp_path,
+            left_catalog_dir=small_sky_object_catalog,
+            left_ra_column="ra",
+            left_dec_column="dec",
+            left_id_column="id",
+            right_ra_column="source_ra",
+            right_dec_column="source_dec",
+            right_id_column="source_id",
+            input_path=small_sky_dir,
+            input_format="csv",
+            metadata_file_path=formats_yaml,
+        )
+
+def test_macauff_args_no_right_catalog_id(
+        small_sky_object_catalog,
+        small_sky_source_catalog,
+        small_sky_dir,
+        formats_yaml,
+        tmp_path
+):
+    with pytest.raises(ValueError, match="right_id_column"):
+        args = MacauffArguments(
+            output_path=tmp_path,
+            output_catalog_name="object_to_source",
+            tmp_dir=tmp_path,
+            left_catalog_dir=small_sky_object_catalog,
+            left_ra_column="ra",
+            left_dec_column="dec",
+            left_id_column="id",
+            right_catalog_dir=small_sky_source_catalog,
+            right_ra_column="source_ra",
+            right_dec_column="source_dec",
+            input_path=small_sky_dir,
+            input_format="csv",
+            metadata_file_path=formats_yaml,
+        )
+
+def test_macauff_args_right_invalid_catalog(
+        small_sky_object_catalog,
+        small_sky_dir,
+        formats_yaml,
+        tmp_path
+):
+    with pytest.raises(ValueError, match="right_catalog_dir"):
+        args = MacauffArguments(
+            output_path=tmp_path,
+            output_catalog_name="object_to_source",
+            tmp_dir=tmp_path,
+            left_catalog_dir=small_sky_object_catalog,
+            left_ra_column="ra",
+            left_dec_column="dec",
+            left_id_column="id",
+            right_catalog_dir=small_sky_dir, # valid directory with files, not a catalog
+            right_ra_column="source_ra",
+            right_dec_column="source_dec",
+            right_id_column="source_id",
+            input_path=small_sky_dir,
+            input_format="csv",
+            metadata_file_path=formats_yaml,
+        )
+
+def test_macauff_args_no_metadata(
```

Comment:
These required argument tests can get tedious to write. I made an iterative checker in the soap tests, if you want to use that model: /tests/hipscat_import/soap/test_soap_arguments.py